src/main/java/com/vinylteam/vinyl/util/impl/VinylUaParser.java

```diff
@@ -4,141 +4,241 @@
+            String currency = priceDetails.substring(priceDetails.indexOf(' ') + 1);
+            Optional<Currency> optionalCurrency = Currency.getCurrency(currency);
+            logger.debug("Got optional with currency from price details {'optionalCurrency':{}, 'priceDetails':{}}", optionalCurrency, priceDetails);
+            return optionalCurrency;
+        }
+        logger.warn("Can't find currency description from price details, returning empty optional {'priceDetails':{}}", priceDetails);
+        return Optional.empty();
+    }
+
+    String getHighResImageLinkFromDocument(Document document) {
+        List<DataNode> scriptDataNodes = document.select(SELECTOR_SCRIPT_WITH_HIGH_RES_IMAGE_LINK).dataNodes();
+        if (scriptDataNodes.isEmpty()) {
+            logger.warn("No script containing high resolution image link by offer link, returning default {'offerLink':{}}", document.location());
+            return "img/goods/no_image.jpg";
+        }
+        String scriptWithHighResImageLink = scriptDataNodes.get(0).getWholeData();
+        logger.debug("Got script containing high resolution image link from page by offer link {'script':{}, 'offerLink':{}}", scriptWithHighResImageLink, document.location());
+        String highResImageLink;
+        int beginIndexOfImageLink = scriptWithHighResImageLink.indexOf("src: '") + "src: '".length();
+        if (beginIndexOfImageLink != -1) {
+            int endIndexOfImageLink = scriptWithHighResImageLink.indexOf('\'', beginIndexOfImageLink);
+            highResImageLink = scriptWithHighResImageLink.substring(beginIndexOfImageLink, endIndexOfImageLink);
+            logger.debug("Got high resolution image link from page by offer link {'highResImageLink':{}, 'offerLink':{}}", highResImageLink, document.location());
+        } else {
+            logger.warn("Can't find image link from page by offer link, returning default {'offerLink':{}}", document.location());
+            highResImageLink = "img/goods/no_image.jpg";
+        }
+        return highResImageLink;
+    }
+
+    String getGenreFromDocument(Document document) {
+        String genre = document.select(SELECTOR_GENRE).text();
+        logger.debug("Got genre from page by offer link {'genre':{}, 'offerLink':{}}", genre, document.location());
+        return genre;
+    }
+
+    private boolean isValid(RawOffer rawOffer) {
+        if (!"".equals(rawOffer.getRelease())) {
+            return true;
+        }
+        if (!"Various Artists".equals(rawOffer.getArtist())) {
+            return true;
+        }
+        if (rawOffer.getPrice() != 0.) {
+            return true;
+        }
+        if (rawOffer.getCurrency().isPresent()) {
+            return true;
+        }
+        if (!"".equals(rawOffer.getGenre())) {
+            return true;
+        }
+        if (!"img/goods/no_image.jpg".equals(rawOffer.getImageLink())) {
+            return true;
+        }
+        return false;
+    }
```

Comment:
Please make it package-private and add test on it
Where are you using this method?
Change it! This method should check all conditions at one time not only one of them. After row "return" method execution is over.